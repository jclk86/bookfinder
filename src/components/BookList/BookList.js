import React, { Component } from "react";
import BookItem from "../BookItem/BookItem";
import "./BookList.css";

export default class BookList extends Component {
  static defaultProps = {
    bookList: []
  };
  render() {
    const bookList = this.props.bookList.map((book, i) => (
      <BookItem {...book} key={book.id} book={book} />
    ));
    return <ul>{bookList}</ul>;
  }
}
