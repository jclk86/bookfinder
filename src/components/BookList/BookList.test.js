import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import BookList from "./BookList";

describe("BookList Component", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<BookList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
