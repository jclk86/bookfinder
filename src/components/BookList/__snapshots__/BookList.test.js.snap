// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList Component renders without crashing 1`] = `<ul />`;
